Source/WTF/wtf/Vector.h

```diff
@@ -1026,6 +1026,9 @@ template<typename T, size_t inlineCapacity, typename OverflowHandler, size_t min
 Vector<T, inlineCapacity, OverflowHandler, minCapacity, Malloc>::Vector(const Vector& other)
     : Base(other.size(), other.size())
 {
+#ifdef __swift__
+    RELEASE_ASSERT_NOT_REACHED_WITH_MESSAGE("C++ copy constructors are not memory safe in Swift. Try to find a way to borrow the object instead.");
+#endif
```

Comment:
I think that committing this is a risk per our discussion in https://github.com/WebKit/WebKit/pull/26438#discussion_r1557971121. Seems like it is an ODR violation, and in practice the linker could choose the swift-compiled instantiation arbitrarily and break callers from C++.